src/base/bittorrent/session.cpp

```diff
@@ -4030,7 +4072,10 @@ void Session::handleTorrentDeleteFailedAlert(const lt::torrent_delete_failed_ale
     const RemovingTorrentData tmpRemovingTorrentData = m_removingTorrents.take(infoHash);
     // libtorrent won't delete the directory if it contains files not listed in the torrent,
     // so we remove the directory ourselves
-    Utils::Fs::smartRemoveEmptyFolderTree(tmpRemovingTorrentData.savePathToRemove);
+    if (tmpRemovingTorrentData.deleteOption == TorrentAndFiles)
+        Utils::Fs::smartRemoveEmptyFolderTree(tmpRemovingTorrentData.savePathToRemove);
+    else if (tmpRemovingTorrentData.deleteOption == TorrentAndFolder)
+        Utils::Fs::removeDirRecursive(tmpRemovingTorrentData.savePathToRemove);
 
     if (p->error) {
         LogMsg(tr("'%1' was removed from the transfer list but the files couldn't be deleted. Error: %2", "'xxx.avi' was removed...")
```

Comment:
Something has to be done with this because it is shown even when `removeDirRecursive` is used. I don't really know what "smartRemoveEmptyFolderTree" does so not sure if it should go in `if (tmpRemovingTorrentData.deleteOption == TorrentAndFiles)`. Please advise.

Reply:
>Something has to be done with this because it is shown even when removeDirRecursive is used. 

I don't think `removeDirRecursive` is guaranteed to succeed doing its job (currently there is no way to know) and this msg could still be needed.

>I don't really know what "smartRemoveEmptyFolderTree" does 

It removes folders and only when they do not contain any files.

Personally I think showing this log msg is not a very a big issue.